File: library/src/test/java/io/ashdavies/rx/rxfirebase/AuthStateFlowableTest.java
package io.ashdavies.rx.rxfirebase;

import com.google.firebase.auth.FirebaseAuth;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import io.reactivex.FlowableEmitter;

import static org.mockito.BDDMockito.then;

@RunWith(MockitoJUnitRunner.class)
public class AuthStateFlowableTest {

  private AuthStateFlowable flowable;

  @Mock FirebaseAuth auth;
  @Mock FlowableEmitter<FirebaseAuth> emitter;

  @Before
  public void setUp() throws Exception {
    flowable = new AuthStateFlowable(emitter);
  }

  @Test
  public void shouldEmitAuthStateChange() throws Exception {
    flowable.onAuthStateChanged(auth);

    then(emitter).should().onNext(auth);
  }

  @Test
  public void shouldRemoveAuthStateListener() throws Exception {
    flowable.cancellable(auth).cancel();

    then(auth).should().removeAuthStateListener(flowable);
  }
}

File: src/test/kotlin/io/ashdavies/rx/rxfirebase/AuthStateFlowableTest.kt
package io.ashdavies.rx.rxfirebase

import com.google.firebase.auth.FirebaseAuth
import com.nhaarman.mockito_kotlin.mock
import com.nhaarman.mockito_kotlin.then
import io.reactivex.FlowableEmitter
import org.junit.jupiter.api.Test

internal class AuthStateFlowableTest {

  private val emitter: FlowableEmitter<FirebaseAuth> = mock()
  private val auth: FirebaseAuth = mock()

  private val flowable = AuthStateFlowable(emitter)

  @Test
  fun `should invoke on next`() {
    flowable.onAuthStateChanged(auth)

    then(emitter)
        .should()
        .onNext(auth)
  }
}
